### ISSUE TITLE
Any plan on support for pure javascript on client side?

### ISSUE THREAD

**Issue by amitamb** [NaturalNode/natural#25]
I did not go too deep to understand if current version can be used on client side as it is. But given the training data needed I guess not.

Is there any plan on making it available to use in browser rather than though node?

Thanks for your efforts. This will be a useful resource in the future.

**Comment by keplersj** [comment]
Sorry for the update from 5-years in the future, but I found a solution for my derivative of this problem with Webpack (through Gatsby).

To solve this I set webpack to resolve `fs` to webpack's [`memory-fs`](https://github.com/webpack/memory-fs) (which works in browser) and null out `webworker-threads`. This has allowed natural to work in browser for me.

Here is a sample `gatsby-node.js` config which accomplishes this:
```js
exports.modifyWebpackConfig = ({ config, stage }) => {
  config.merge({
    resolve: {
      alias: {
        fs: "memory-fs"
      }
    }
  });

  config.loader("null", {
    test: /webworker-threads/,
    loader: "null-loader"
  });

  return config;
};
```

**Comment by newvicklee** [comment]
Found a workaround in case anyone is wondering. I've only tested this with the Tokenizer module at the moment. 
Go to /node_modules/natural/lib/natural/index.js and comment out all modules except the tokenizer ones (or whichever ones you want to use). There are some modules that just won't work on the client side (those that use node specific functions like fs and readFileSync).
Install browserify and bundle up your client side JS code that uses NaturalNode. With only the tokenizer module, it comes out to under 200kB.
